core/storage/suggestion/gae_models_test.py

```diff
@@ -2994,3 +2994,1639 @@ def test_get_model_association_to_user(self) -> None:
+                self.TOPIC_IDS_WITH_TRANSLATION_REVIEWS),
+            reviewed_translations_count=self.REVIEWED_TRANSLATIONS_COUNT,
+            accepted_translations_count=self.ACCEPTED_TRANSLATIONS_COUNT,
+            accepted_translations_with_reviewer_edits_count=(
+                self.ACCEPTED_TRANSLATIONS_WITH_REVIEWER_EDITS_COUNT),
+            accepted_translation_word_count=(
+                self.ACCEPTED_TRANSLATION_WORD_COUNT),
+            rejected_translations_count=(
+                self.REJECTED_TRANSLATIONS_COUNT),
+            first_contribution_date=datetime.datetime.utcnow(),
+            last_contribution_date=datetime.date(2023, 4, 2)
+        ).put()
+
+        # Check for decreasing order of reviewed cards(default)
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationReviewerTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=None,
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_3')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for increasing order of reviewed cards.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationReviewerTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingReviewedTranslations'],
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_1')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for decreasing order of last activity.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationReviewerTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'DecreasingLastActivity'],
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_1')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for increasing order of last activity.
+        sorted_results, next_offset, more = (
+            suggestion_models.TranslationReviewerTotalContributionStatsModel
+            .fetch_page(
+            page_size=1,
+            offset=0,
+            sort_by=constants.CD_ADMIN_STATS_SORT_OPTIONS[
+                    'IncreasingLastActivity'],
+            last_activity=None,
+            language_code='es'
+            ))
+        self.assertEqual(sorted_results[0].id, 'model_2')
+        self.assertTrue(more)
+        self.assertEqual(next_offset, 1)
+
+        # Check for language filter.
```

Comment:
Same notes as before re splitting filtering to a separate test. Also note that last_activity is misspelled below.

Ditto for the other two models.

Reply:
I think, the current test cases covers both sorting and filtering and we need to ensure that both works coherently as well. Separating the tests won't ensure that sorting and filtering works fine even when applied together. Also the same question as above by splitting you mean to split in two different test classes?

Reply:
I would suggest having different test methods (not classes) -- one for testing sorting, one for testing filtering, and one for testing the combo. Also, name your test methods clearly -- not just "test_method_name". See https://github.com/oppia/oppia/wiki/Backend-tests#guidelines-for-writing-good-tests for guidance.

Reply:
PTAL at new tests.